lab_3_generate_by_ngrams/main.py

```diff
@@ -121,6 +177,16 @@ def decode(self, encoded_corpus: tuple[int, ...]) -> Optional[str]:
         In case of corrupt input arguments, None is returned.
         In case any of methods used return None, None is returned.
         """
+        if not isinstance(encoded_corpus, tuple) or\
+                len(encoded_corpus) == 0:
+            return None
+        decoded_corpus = self._decode(encoded_corpus)
+        if decoded_corpus is None:
+            return None
+        decoding = self._postprocess_decoded_text(decoded_corpus)
+        if decoding is None:
```

Comment:
Is this check necessary? Consider directly returning `decoding` as this function can also return `None`

Btw, -ing form in the variable name is confusing.

Reply:
fixed